### PR TITLE
Improve eslint

### LINKED ISSUE #1525: TS + ESLint upgrades
I'd like to upgrade typescript + eslint to be the latest and update the configuration for the latest. While initially I'd try to get it set up such that changes to existing code aren't required (or are minimal), I'd later like to introduce new eslint rules that improve the types across the app to prevent unexpected behaviors and add code clarity.

Sound good?

### PR DESCRIPTION
**Pull request by corasaurus-hex**
## What has Changed?

This PR is an overhaul of of linting for the project while temporarily keeping it compatible with the current code. The goal is to then turn rules on a few at a time and contribute them as subsequent PRs.

For the reviewer, the general things this PR does are:

1. Removes tslint, which was [deprecated in 2019](https://blog.palantir.com/tslint-in-2019-1a144c2317a9).
2. Upgrades the typescript package.
3. Upgrades some eslint packages.
4. Removes some other eslint packages.
5. Adds updated configurations for eslint and typescript.
6. Adds eslint as a step in in CI.
7. Changes `cp` commands in the CI config to use arguments that should work cross-platform.

Fixes #1525

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [x] Tests
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
     - [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [~] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [x] Created the issue I am fixing/addressing, if it was not present.
- [x] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [x] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) or run `npm run eslint`).

Ping @pez, @bpringe

### CODE REVIEW

**Comment by corasaurus-hex** [comment]
@PEZ and @bpringe : this is ready for review when you get a chance 😎

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Changes to Calva.
 
 ## [Unreleased]
```
Comment (PEZ) [inline CHANGELOG.md]: I think that the script we have for picking out the changes is still picky about no new-line here.

**Review by PEZ — CHANGES_REQUESTED** [review]
We should fix so that we don't need that newline in the changelog, but anyway. =)

File: CHANGELOG.md (hunk 1 above)
Comment (corasaurus-hex) [reply CHANGELOG.md]: Fixed!!

**Review by PEZ — APPROVED** [review]
Thanks!

**Comment by bpringe** [comment]
> We should fix so that we don't need that newline in the changelog, but anyway. =)

https://github.com/BetterThanTomorrow/calva/issues/1534 👍

**Pull request merged**